temboardui/autossl.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import ssl
+
+from tornado.httpserver import HTTPServer
+from tornado.httputil import HTTPHeaders
+
+
+logger = logging.getLogger(__name__)
+
+
+def is_http_request(error):
+    if error.args[0] != ssl.SSL_ERROR_SSL:
+        return False
+
+    if getattr(error, 'reason', None) != 'HTTP_REQUEST':
+        return False
+
+    return True
+
+
+def parse_http_headers(payload):
+    lines = payload.splitlines(True)
+    # Drop start line
+    lines.pop(0)
+    # Drop contents
+    if '' in lines:
+        lines[:] = lines[:lines.index('')]
+    return HTTPHeaders.parse(''.join(lines))
+
+
+def redirect_response(headers):
+    # We ignore request path and redirect to /.
+    new_url = 'https://%s/' % headers.get('Host', '0.0.0.0:8888')
+    logger.info("Redirecting client to %s.", new_url)
+    return (
+        "HTTP/1.0 301 Redirect\r\n"
+        "Location: %(new_url)s\r\n"
+        "\r\n" % dict(new_url=new_url)
+    )
+
+
+class AutoHTTPSServer(HTTPServer):
+    # Naïve HTTP server redirecting HTTP requests to HTTPS.
+
+    def handle_stream(self, stream, address):
+        stream.socket.setblocking(1)
```

Comment:
@julmon I'm not happy with that. I will subclass SSHIOStream to implement async handshake.